src/main/java/org/ngafid/common/terrain/TerrainCache.java

```diff
@@ -1,31 +1,52 @@
 package org.ngafid.common.terrain;
 
 import java.nio.file.NoSuchFileException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
 
 
 public final class TerrainCache {
-
-    private static final SRTMTile[][] TILES = new SRTMTile[180][360];
     public static final String TERRAIN_DIRECTORY;
+    private static int MAX_CACHE_SIZE;
+    private static final Logger LOG = Logger.getLogger(TerrainCache.class.getName());
+    
+
+    // Load factor is arbitrary because we always remove the eldest entry
+    private static final Map<Coordinate, SRTMTile> TILE_CACHE = new ConcurrentHashMap<>(new LinkedHashMap<>(MAX_CACHE_SIZE, 0.75f, true) {
```

Comment:
This constructor to ConcurrentHashMap creates a copy of the mappings in the supplied map -- I don't think this will work.

It looks like there is no standard `LinkedConcurrentHashMap` and using a lock to access the map each and everytime is WEAK. You will have to implement the cache-order thing yourself I fear -- enjoy.